tensorflow/core/kernels/fixed_length_record_reader_op.cc

```diff
@@ -62,14 +64,20 @@ class FixedLengthRecordReader : public ReaderBase {
 
   Status ReadLocked(string* key, string* value, bool* produced,
                     bool* at_end) override {
-    if (input_buffer_->Tell() >= file_pos_limit_) {
+    if (input_buffer_->Tell() >= file_pos_limit_ ||
+        input_buffer_->Tell() + record_bytes_ > file_pos_limit_) {
```

Comment:
Ah, thanks for the explanation.  I think you meant that after the first three successful reads, the `Tell()` cursor will be at `|` in `012345|6`.  Please add this explanation for `hop_bytes > 0` and this example in a comment.